### PR TITLE
V2 - better modules support

### PR DESCRIPTION
**Pull request by jorisre**
First of all, huge thank to the [preact](https://github.com/preactjs/preact) hard work. This PR is heavily inspired of their modules structure.

Good news, no breaking changes ! users can import resolvers as usual 🎉 

- move resolvers into separate folders
- add exports map in package.json + support node 13 _(thank to preact)_
- build: use microbundle + severals updates
- test: update config
- remove IE11
- use semantic-release default config

Tested with Next.js and CRA with success ✅  ! Others formats will be tested after first beta release

Solves https://github.com/react-hook-form/resolvers/issues/71 & https://github.com/react-hook-form/resolvers/issues/100

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,19 +136,14 @@
   "peerDependencies": {
     "react-hook-form": ">=6.6.0"
   },
-  "release": {
```
Comment (jorisre) [inline package.json:91]: By removing this, this allows us to use semantic-release's default config, so master, beta branches will be automatically published.

In our case, beta will create 2.0.0-beta.X version

https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration#pre-release-branches

**Comment by bluebill1049** [comment]
This is super exciting!

**Comment by jorisre** [comment]
@bluebill1049 PR ready for review 🙂

**Comment by bluebill1049** [comment]
Screen Shot 2021-01-04 at 8 40 41 pm

Wow, this is cool!

File: joi/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  "description": "React Hook Form validation resolver: joi",
+  "main": "dist/joi.js",
+  "module": "dist/joi.module.js",
+  "umd:main": "dist/joi.umd.js",
+  "source": "src/index.ts",
+  "types": "dist/index.d.ts",
+  "license": "MIT",
+  "peerDependencies": {
+    "react-hook-form": ">=6.6.0",
+    "@hookform/resolvers": "^1.0.0"
```
Comment (bluebill1049) [inline joi/package.json:15]: dependent itself?

File: package.json [hunk 3]
```diff
@@ -1,36 +1,89 @@
...
+      "umd": "./vest/dist/vest.umd.js",
+      "import": "./vest/dist/vest.mjs",
+      "require": "./vest/dist/vest.js"
+    },
+    "./superstruct": {
+      "browser": "./superstruct/dist/superstruct.module.js",
+      "umd": "./superstruct/dist/superstruct.umd.js",
+      "import": "./superstruct/dist/superstruct.mjs",
+      "require": "./superstruct/dist/superstruct.js"
+    },
```
Comment (bluebill1049) [inline package.json:47]: nice!

File: package.json [hunk 4]
```diff
@@ -1,36 +1,89 @@
...
-    "build": "tsc && rollup -c",
-    "lint": "eslint '**/*.{js,ts}'",
-    "lint:fix": "yarn lint -- --fix",
-    "lint:types": "tsc --noEmit",
-    "test": "jest --runInBand",
-    "test:watch": "yarn test --watchAll --coverage"
+    "prepare": "run-s build && check-export-map",
+    "build": "npm-run-all --parallel build:*",
+    "build:src": "microbundle build",
+    "build:zod": "microbundle --cwd zod --globals '@hookform/resolvers=hookformResolvers'",
```
Comment (bluebill1049) [inline package.json:76]: That's new to me, `--globals`

**Review by bluebill1049 — APPROVED** [review]
This is too good! Does this need a breaking change? if not maybe we can keep it under v1, until v2 with breaking changes (those new attributes, arguments, and functionalityies)?

File: joi/package.json (hunk 2 above)
Comment (jorisre) [reply joi/package.json:15]: Yes, to import `convertArrayToPathName`, inspiration from [preact](https://github.com/preactjs/preact/blob/master/hooks/package.json#L21)

File: package.json (hunk 4 above)
Comment (jorisre) [reply package.json:76]: It specify variable name for external imports in umd/iife bundles

**Comment by jorisre** [comment]
> This is too good! Does this need a breaking change? if not maybe we can keep it under v1, until v2 with breaking changes (those new attributes, arguments, and functionalityies)?

I suggest to publish a V2 beta because there are lot of changes (project structure, bundles etc...) And we can advise users to try beta and collect feedbacks. What do you think ?
I’m thinking about a powerful v2 without any breaking changes for developers :) 🤞🏻

**Comment by bluebill1049** [comment]
> > This is too good! Does this need a breaking change? if not maybe we can keep it under v1, until v2 with breaking changes (those new attributes, arguments, and functionalityies)?
> 
> I suggest to publish a V2 beta because there are lot of changes (project structure, bundles etc...) And we can advise users to try beta and collect feedbacks. What do you think ?
> I’m thinking about a powerful v2 without any breaking changes for developers :) 🤞🏻

sounds good to me! The only thing to beware that their argument of resolver may change with v7, we didn't document the third argument in the readme tho, just in case someone is using it.

File: joi/package.json (hunk 2 above)
Comment (bluebill1049) [reply joi/package.json:15]: oh cool! 👍

**Pull request merged**